src/asl.ts

```diff
@@ -2774,37 +2772,64 @@ export class ASL {
         const ref = expr.lookup();
         if (ref === undefined) {
           throw new Error(`unresolved identifier: ${expr.name}`);
-        } else if (isParameterDecl(ref)) {
-          if (ref.parent !== predicate) {
-            throw new Error(
-              "cannot reference a ParameterDecl other than those in .filter((item, index) =>) in a JSONPath filter expression"
-            );
-          }
-          if (ref === ref.parent.parameters[0]) {
-            return "@";
-          } else if (ref === ref.parent.parameters[1]) {
-            throw new Error(
-              "the 'index' parameter in a .filter expression is not supported"
-            );
-          } else {
+        }
+        return resolveRef(ref);
+        function resolveRef(ref: Decl): string {
+          if (isParameterDecl(ref)) {
+            if (ref.parent !== predicate) {
+              throw new Error(
+                "cannot reference a ParameterDecl other than those in .filter((item, index) =>) in a JSONPath filter expression"
+              );
+            }
+            if (ref === ref.parent.parameters[0]) {
+              return "@";
```

Comment:
what is the `@` for? quick comment?

Reply:
Most of this was your code that I just wrapped in a new method, so I am not 100% sure how it works. I think its the `@` in the json path filter expression.